app/repositories/InvitationsCacheRepository.scala

```diff
@@ -106,21 +106,43 @@ class InvitationsCacheRepository @Inject()(
   }
 
   private val expireAt = "expireAt"
-  private val createdIndexName = "dataExpiry"
+  private val dataExpiry = "dataExpiry"
   private val expireAfterSeconds = "expireAfterSeconds"
   private val inviteePsaIdKey = "inviteePsaId"
   private val pstrKey = "pstr"
-  private val compoundIndexName = "inviteePsaId_Pstr"
+  private val uniqueInvitation = "inviteePsaId_Pstr"
   private val pstrIndexName = "pstr"
 
-  ensureIndex(fields = Seq(expireAt), indexName = createdIndexName, ttl = Some(ttl)) andThen {
-    case _ => ensureIndex(fields = Seq(inviteePsaIdKey, pstrKey), indexName = compoundIndexName, isUnique = true)
-  } andThen {
-    case _ => ensureIndex(fields = Seq(pstrKey), indexName = pstrIndexName)
+  (for {
+    _ <- checkIndexTtl(dataExpiry, Some(ttl))
+    _ <- ensureIndex(fields = Seq(expireAt), indexName = dataExpiry, ttl = Some(ttl))
+    _ <- ensureIndex(fields = Seq(inviteePsaIdKey, pstrKey), indexName = uniqueInvitation, isUnique = true)
+    _ <- ensureIndex(fields = Seq(pstrKey), indexName = pstrIndexName)
+  } yield {
+    ()
+  }) recoverWith {
+    case t: Throwable => Future.successful(Logger.error(s"Error ensuring indexes on collection ${collection.name}", t))
   } andThen {
     case _ => CollectionDiagnostics.logCollectionInfo(collection)
   }
 
+  private def checkIndexTtl(indexName: String, ttl: Option[Int]): Future[Unit] = {
```

Comment:
This bit of code is in both InvitationsCacheRepository and PensionAdministratorCacheRepository - perhaps it could be placed in a common area? Maybe a parent trait?